### PR TITLE
HIP 29: M-of-N Multisig

### PR DESCRIPTION
**Pull request by xandkar**
Rendered: https://github.com/helium/HIP/blob/sk/multisig/0029-multisignature-keys.md

### CODE REVIEW

File: 0029-multisig-addresses.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+- Category: economic, technical
+- Original HIP PR:
+- Tracking Issue:
+
+# Summary
+[summary]: #summary
+
+Add  a new type of public key and a new type of signature, which are composites
+of the scalar public keys and signatures combined with a specification for the
+minimal subset of signatures required to make a transaction.
```
Comment (abhay) [inline 0029-multisig-addresses.md:14]:
```suggestion
Helium Hotspot owners and HNT holders have different ways to organize
assets. This HIP proposes a multi-signature key whihch can be used to
authorize a transaction on the Helium blockchain. This key is a composite
of the scalar keys which can be combined to generate a minimal subset of 
signatures required to make a valid transaction.
```

File: 0029-multisig-addresses.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+# HIP 29: Multisig transactions
```
Comment (abhay) [inline 0029-multisig-addresses.md:1]:
```suggestion
# HIP 29: Multi-signature Keys
```

Also update the filename?

File: 0029-multisignature-keys.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+- Start Date: 2021-04-19
+- Category: technical
+- Original HIP PR: [#154](https://github.com/helium/HIP/pull/154)
+- Tracking Issue:  <!-- TODO -->
+
+# Summary
+[summary]: #summary
+
+Helium Hotspot owners and HNT holders have different ways to organize assets.
+This HIP proposes a multi-signature key whihch can be used to authorize a
```
Comment (Vagabond) [inline 0029-multisignature-keys.md:13]:
```suggestion
This HIP proposes a multi-signature key which can be used to authorize a
```

File: 0029-multisignature-keys.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    | variable length | variable length  |
+    +-----------------+------------------+
+
+Due to variable lengths of the components, parsing of the multi-signature
+requires knowledge of:
+1. `N` from the corresponding multi-pub-key;
+2. length of each scalar key type (type itself is encoded, but the length
+   corresponding to type is not, so must be known in advance).
+
+### verification
```
Comment (abhay) [inline 0029-multisignature-keys.md:95]:
```suggestion
### Verification
```

File: 0029-multisignature-keys.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
+# HIP 29: Multi-signature Keys
+
+- Author(s): @xandkar, Helium Team; @Vagabond, Helium Team.
```
Comment (abhay) [inline 0029-multisignature-keys.md:3]:
```suggestion
- Author(s): [@xandkar](http://github.com/xandkar), Helium Team; [@Vagabond](http://github.com/Vagabond), Helium Team.
```

File: 0029-multisignature-keys.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+- Start Date: 2021-04-19
+- Category: technical
```
Comment (abhay) [inline 0029-multisignature-keys.md:5]:
```suggestion
- Category: Technical
```

**Comment by abhay** [comment]
RFC

**Review by abhay — APPROVED** [review]

**Comment by jamiew** [comment]
Draft looks good to me. Will merge for discussion. This will be HIP 29

**Comment by jamiew** [comment]
This HIP draft has been accepted and merged for discussion. Please direct comments & feedback to the tracking issue: https://github.com/helium/HIP/issues/157

**Comment by Jamill-hallak** [comment]
I think when you want to make M-of-N multisig , we should build some functions to get a special private key from other basic private keys .so is the code of Helium have some functions to build a special private key from other basic private keys?
if yes: can you show me the resource code of it?
if not: I can build some functions like this, but here I need to know all about the standard building of helium's private key.

**Pull request merged**